spec/system/sign_in_redirect_spec.rb

```diff
@@ -0,0 +1,66 @@
+require "rails_helper"
+
+RSpec.describe "Sign-in redirect", type: :system do
+  let!(:sakes) { FactoryBot.create_list(:sake, 3) }
+  let(:id) { sakes[0].id }
+  let!(:user) { FactoryBot.create(:user) }
+
+  before do
+    driven_by(:rack_test)
+    sign_out(user)
+  end
+
+  context "when non-signed-in user tries to edit-sake" do
+    before do
+      visit sake_path(id)
+      find(:test_id, "edit-#{id}").click
+    end
+
+    it "redirects user to new_user_session_path" do
+      expect(page).to have_current_path new_user_session_path
+    end
+
+    it "redirect to edit_sake_path after user sign-in" do
+      fill_sign_column(user)
+      expect(page).to have_current_path edit_sake_path(id)
+    end
+  end
+
+  context "when non-signed-in user tries to new-sake" do
+    before do
+      visit sakes_path
+      find(:test_id, "new-sake").click
+    end
+
+    it "redirects user to new_user_session_path" do
+      expect(page).to have_current_path new_user_session_path
+    end
+
+    it "redirects user to new_sake_path after user sign-in" do
+      fill_sign_column(user)
+      expect(page).to have_current_path new_sake_path
+    end
+  end
+
+  context "when user presses sign-in button on sake show page" do
```

Comment:
`user presses`だけど、これだとuserの目的がボタンを押すことと読み取れる。
でもこのテストは、「酒showページでユーザがログインしたときの動作」なのでuserの目的はログイン。

```suggestion
  context "when user sign in from sake show page" do
```

Reply:
when user signs in from header button on sake show page
「酒Showページのヘッダーボタンでユーザーがログインしたとき」にします。
理由：酒Showページからログインする方法が2つ（ヘッダーのサインインを押すor酒編集ボタンを押す）あって、どちらの方法かによって挙動が異なるため。

Reply:
Fixed in 5c30803